assets/js/admin/block-extension.js

```diff
@@ -15,6 +16,29 @@ const BlockEdit = (props) => {
 	const { attributes, setAttributes } = props;
 	const { selectedPosts, query, namespace } = attributes;
 
+	const previousPostTypeRef = useRef(query.postType);
+
+	useEffect(() => {
+		if (query.postType !== previousPostTypeRef.current) {
+			const newAttributes = { ...attributes };
+
+			if (newAttributes.query && newAttributes.query.include) {
+				delete newAttributes.query.include;
+			}
+
+			newAttributes.selectedPosts = [];
+			newAttributes.query = {
+				...newAttributes.query,
+				orderBy: 'date',
+				perPage: newAttributes.query.perPage,
```

Comment:
```suggestion
				perPage: newAttributes.query.perPage,
				include: [],
```

Reply:
This prevents a single post from showing by default when switching between post types.